test/modules/load-gen/LoadGenSenderBase.cs

```diff
@@ -23,7 +23,7 @@ public abstract class LoadGenSenderBase
             this.Logger = Preconditions.CheckNotNull(logger, nameof(logger));
             this.Client = Preconditions.CheckNotNull(moduleClient, nameof(moduleClient));
             this.BatchId = Preconditions.CheckNotNull(batchId, nameof(batchId));
-            this.TrackingId = Preconditions.CheckNonWhiteSpace(trackingId, nameof(trackingId));
+            this.TrackingId = trackingId;
```

Comment:
Even tracking id is not mandatory, it is better to not null.  btw, just a little sharing, do you know "??=" operator; read https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator if interested.
```suggestion
            this.TrackingId = trackingId ?? string.Empty;
```